packages/flutter/lib/src/services/asset_bundle.dart

```diff
@@ -88,15 +69,41 @@ class MojoAssetBundle extends AssetBundle {
     return new String.fromCharCodes(new Uint8List.view(data.buffer));
   }
 
-  Future<core.MojoDataPipeConsumer> load(String key) async {
-    return (await _bundle.ptr.getAsStream(key)).assetData;
-  }
-
   Future<String> loadString(String key) {
     return _stringCache.putIfAbsent(key, () => _fetchString(key));
   }
 }
 
+class MojoAssetBundle extends CachingAssetBundle {
+  MojoAssetBundle(this._bundle);
+
+  factory MojoAssetBundle.fromNetwork(String relativeUrl) {
+    AssetBundleProxy bundle = new AssetBundleProxy.unbound();
+    _fetchAndUnpackBundle(relativeUrl, bundle);
+    return new MojoAssetBundle(bundle);
+  }
+
+  static Future _fetchAndUnpackBundle(String relativeUrl, AssetBundleProxy bundle) async {
+    core.MojoDataPipeConsumer bundleData = (await fetchUrl(relativeUrl)).body;
+    AssetUnpackerProxy unpacker = new AssetUnpackerProxy.unbound();
+    shell.connectToService("mojo:asset_bundle", unpacker);
+    unpacker.ptr.unpackZipStream(bundleData, bundle);
+    unpacker.close();
+  }
+
+  AssetBundleProxy _bundle;
```

Comment:
FWIW, this was all existing code.  I extracted a few methods out into a base class but didn't change anything else.